vocode/streaming/synthesizer/azure_synthesizer.py

```diff
@@ -358,11 +359,15 @@ def get_message_up_to(
         word_boundary_event_pool: WordBoundaryEventPool,
     ) -> str:
         events = word_boundary_event_pool.get_events_sorted()
-        # for event in events:
-        #     if event["audio_offset"] > seconds:
-        #         ssml_fragment = ssml[: event["text_offset"]]
-        #         # TODO: this is a little hacky, but it works for now
-        #         return ssml_fragment.split(">")[-1]
+        message_index: int = events[0]["text_offset"]  # Start of the message
+        for index, event in enumerate(events):
+            if event["audio_offset"] > seconds:
+                current_index = events[index - 1]["text_offset"] + len(
+                    events[index - 1]["text"]
+                )
+                message_up_to = ssml[message_index:current_index]
```

Comment:
nit: you don't need to change it here but i think you can make it more readable by using a different helper variable. The definition of `current_index` is hard to understand, especially cuz the name is vague and it's hard to think of a more precise one. I would have done something like
```suggestion
                message_start = events[index - 1]["text_offset"]
                message_length = len(events[index - 1]["text"])
                return ssml[message_start:message_start+message_length]
```